proxy-robots-module/src/main/java/orwell/proxy/ProxyRobots.java

```diff
@@ -61,37 +64,35 @@ private void connectToServer() {
     }
 
     /**
-     * This instantiates Tanks objects from a configuration It only set up the
-     * tanksInitializedMap
+     * This instantiates Robots objects from a configuration. It only sets up the
+     * RobotsMap
      */
-    protected void initializeTanksFromConfig() {
-        for (final ConfigTank configTank : configRobots.getConfigRobotsToRegister()) {
-            final Camera camera = new Camera(configTank.getConfigCamera().getIp(),
-                    configTank.getConfigCamera().getPort());
-            //TODO Improve initialization of setImage to get something meaningful
-            //from the string (like an actual picture)
-            final Tank tank = new Tank(configTank.getBluetoothName(),
-                    configTank.getBluetoothID(), camera, configTank.getImage());
-            logback.info("Temporary routing ID: " + configTank.getTempRoutingID());
-            tank.setRoutingId(configTank.getTempRoutingID());
-            this.robotsMap.add(tank);
+    protected void initializeRobotsFromConfig() {
+        for (final IConfigRobot configRobot : configRobots.getConfigRobotsToRegister()) {
+            final IRobot robot = robotFactory.getRobot(configRobot);
+            if (null == robot) {
+                logback.error("Robot not initialized. Skipping it for now.");
+            } else {
+                logback.info("Temporary routing ID: " + robot.getRoutingId());
+                robot.setRoutingId(configRobot.getTempRoutingID());
+                this.robotsMap.add(robot);
+            }
         }
 
         logback.info("All " + this.robotsMap.getRobotsArray().size()
-                + " tank(s) initialized");
+                + " robot(s) initialized");
     }
 
     protected void connectToRobots() {
         for (final IRobot robot : robotsMap.getNotConnectedRobots()) {
-            robot.connectToDevice();
+            robot.connect();
         }
     }
 
     protected void sendRegister() {
         for (final IRobot robot : robotsMap.getConnectedRobots()) {
```

Comment:
It is not just here, but the content of the loop could be extracted in a separate method.